### PR TITLE
e2e operator scaling test

### PR DESCRIPTION
**Pull request by katexochen**
### Proposed change(s)

- Add an e2e scaling test to verify the operator works as expected.

### CODE REVIEW

File: .github/actions/e2e_autoscaling/action.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  kubeconfig:
+    description: "The kubeconfig of the cluster to test."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    # This action assumes that the cluster is in an ready state, with all nodes joined and ready.
+
+    - name: Dertermine number of workers in cluster
```
Comment (malt3) [inline .github/actions/e2e_autoscaling/action.yml:14]:
```suggestion
    - name: Determine number of workers in cluster
```

File: .github/actions/e2e_test/action.yml [hunk 2]
```diff
@@ -36,13 +39,23 @@ inputs:
   azureResourceGroup:
     description: "The resource group to use"
     required: false
-  sonobuoyTestSuiteCmd:
-    description: "Which tests should be run? Check README for guidance!"
+  test:
+    description: "The test to run. Can currently be on of [sonobuoy, autoscaling]."
```
Comment (malt3) [inline .github/actions/e2e_test/action.yml:43]:
```suggestion
    description: "The test to run. Can currently be one of [sonobuoy, autoscaling]."
```

**Review by malt3 — APPROVED** [review]
LGTM. Just small nits.

File: .github/actions/e2e_test/action.yml (hunk 2 above)
Comment (katexochen) [reply .github/actions/e2e_test/action.yml:43]: Ahhrg, there are strings missing in the list, too.

File: .github/workflows/e2e-test-manual-macos.yml [hunk 3]
```diff
@@ -84,7 +88,7 @@ jobs:
     steps:
       - name: Install the basics
         shell: bash
-        run: brew install coreutils
+        run: brew install coreutils kubectl bash
```
[the remark this answers is not in the document]
Comment (katexochen) [reply .github/workflows/e2e-test-manual-macos.yml:91]: We discussed to leave it as is for now, as errors caused by this should be relatively easy to identify. We considered using https://github.com/Azure/setup-kubectl, but has the same problem with our lowest supported version. GitHub's Ubuntu image has the newest version of kubectl installed (non-macos).

**Pull request merged**